compiler/rustc_mir_transform/src/errors.rs

```diff
@@ -62,9 +64,27 @@ impl<'sess, G: EmissionGuarantee> IntoDiagnostic<'sess, G> for RequiresUnsafe {
             handler.struct_diagnostic(crate::fluent_generated::mir_transform_requires_unsafe);
         diag.code(rustc_errors::DiagnosticId::Error("E0133".to_string()));
         diag.set_span(self.span);
-        diag.span_label(self.span, self.details.label());
-        diag.note(self.details.note());
-        let desc = handler.eagerly_translate_to_string(self.details.label(), [].into_iter());
+        diag.span_label(self.span, self.details.clone().label());
+        if let Some(help) = self.help {
+            diag.help(help);
+        }
+
+        match self.details.violation {
```

Comment:
The outer match is not needed here, you can just collapse it to a single `if let`. If it's None, you just do what you do in the `_` arm.

Reply:
This was the old code

Reply:
It's still part of the diff though? Maybe you haven't added these changes to git yet?

Reply:
Ohh you meant what I am asking you to do reflects the older state of a PR, and you were asked to change it? I don't think so. The older state also had a match, just no if let inside. Now you are doing both. Ideally you'd just have an if let, like:

```Rust
if let Some(note_missing_features) = self.details.violation.note_missing_features() {
    diag.note(note_missing_features);
} else {
    diag.note(self.details.clone().note());
}
```

The `note_missing_features` function will already return `None` if it's not a `CallToFunctionWith`, we just use this fact here.

My demand is more or less the same thing I asked about in march: https://github.com/rust-lang/rust/pull/109710#discussion_r1153826985

If you need more clarifications, please ask!